js/apps/system/_admin/aardvark/APP/frontend/js/views/databaseView.js

```diff
@@ -1,513 +0,0 @@
-          window.modalView.createTextEntry(
-            'new-replication-factor',
-            'Replication factor',
-            dbDefaultProperties.replicationFactor,
-            'Numeric value. Must be at least 1. Total number of copies of the data in the cluster',
-            'Default replication factor',
-            false,
-            [
-              {
-                rule: Joi.string().allow('').optional().regex(/^([1-9]|10)$/),
-                msg: 'Must be a number between 1 and 10.'
-              }
-            ]
-          )
-        );
-
-        tableContent.push(
-          window.modalView.createTextEntry(
-            'new-write-concern',
-            'Write concern',
-            dbDefaultProperties.writeConcern,
-            'Numeric value. Must be at least 1. Must be smaller or equal compared to the replication factor. Total number of copies of the data in the cluster that are required for each write operation. If we get below this value the collection will be read-only until enough copies are created.',
-            'Default write concern',
-            false,
-            [
-              {
-                rule: Joi.string().allow('').optional().regex(/^([1-9]|10)$/),
-                msg: 'Must be a number between 1 and 10. Has to be smaller or equal compared to the replicationFactor.'
-              }
-            ]
-          )
-        );
-      }
-
-      // OneShard
-      //if enterprise
-      if (window.App.isCluster && frontendConfig.isEnterprise && !frontendConfig.forceOneShard) {
-        var sharding = [ { value : "",
-                           label : "Sharded"
-                         },
-                         { value : "single",
-                           label : "OneShard"
-                         }
-                       ];
-
-        tableContent.push(
-          window.modalView.createSelectEntry(
-            'newSharding',
-            'Sharding',
-            'flexible',
-            'Selects the default sharding setup for new collections in this database.\n*Sharded* means that collections can have any number of shards, and that shards will be randomly distributed to database servers.\n*OneShard* means that all collections in the database will have only have a single but are all distributed to the same database server. This enables performance optimizations, e.g. pushing down entire queries to database servers. Use *OneShard* when it is known that the dataset fits on a single instance, and *Sharded* for larger datasets.',
-            sharding
-          )
-        );
-      }
-
-      // User Set-UP
-      var users = [];
-      window.App.userCollection.each(function (user) {
-        users.push({
-          value: user.get('user'),
-          label: user.get('user')
-        });
-      });
-
-      tableContent.push(
-        window.modalView.createSelectEntry(
-          'newUser',
-          'Username',
-          this.users !== null ? this.users.whoAmI() : 'root',
```

Comment:
there seems to be a default value for the user -- current user, or root. Have we handled this in the new react view, or is it not required?

Reply:
It's possible to do what we want, we will need to expose an onChange from an inner component to support it. For now, if you think your changes are good, it works for me! We can add the precise behaviour in the future